src/main/flight/altitude.c

```diff
@@ -203,122 +210,104 @@ int32_t calculateAltHoldThrottleAdjustment(int32_t vel_tmp, float accZ_tmp, floa
 
 void calculateEstimatedAltitude(timeUs_t currentTimeUs)
 {
-    static timeUs_t previousTimeUs;
-    uint32_t dTime;
-    int32_t baroVel;
-    float dt;
-    float vel_acc;
-    int32_t vel_tmp;
-    float accZ_tmp;
-    static float accZ_old = 0.0f;
-    static float vel = 0.0f;
-    static float accAlt = 0.0f;
-    static int32_t lastBaroAlt;
-
-#ifdef SONAR
-    int32_t sonarAlt = SONAR_OUT_OF_RANGE;
-    static int32_t baroAlt_offset = 0;
-    float sonarTransition;
-#endif
-
-    dTime = currentTimeUs - previousTimeUs;
-    if (dTime < BARO_UPDATE_FREQUENCY_40HZ)
+    static timeUs_t previousTimeUs = 0;
+    const uint32_t dTime = currentTimeUs - previousTimeUs;
+    if (dTime < BARO_UPDATE_FREQUENCY_40HZ) {
         return;
-
+    }
     previousTimeUs = currentTimeUs;
 
+    static float vel = 0.0f;
+    static float accAlt = 0.0f;
+
+    int32_t baroAlt = 0;
 #ifdef BARO
-    if (!isBaroCalibrationComplete()) {
-        performBaroCalibrationCycle();
-        vel = 0;
-        accAlt = 0;
+    if (sensors(SENSOR_BARO)) {
+        if (!isBaroCalibrationComplete()) {
+            performBaroCalibrationCycle();
+            vel = 0;
+            accAlt = 0;
+        } else {
+            baroAlt = baroCalculateAltitude();
+            estimatedAltitude = baroAlt;
+        }
     }
-
-    baro.BaroAlt = baroCalculateAltitude();
-#else
-    baro.BaroAlt = 0;
 #endif
 
 #ifdef SONAR
-    sonarAlt = sonarRead();
-    sonarAlt = sonarCalculateAltitude(sonarAlt, getCosTiltAngle());
-
-    if (sonarAlt > 0 && sonarAlt < sonarCfAltCm) {
-        // just use the SONAR
-        baroAlt_offset = baro.BaroAlt - sonarAlt;
-        baro.BaroAlt = sonarAlt;
-    } else {
-        baro.BaroAlt -= baroAlt_offset;
-        if (sonarAlt > 0  && sonarAlt <= sonarMaxAltWithTiltCm) {
+    if (sensors(SENSOR_SONAR)) {
+        int32_t sonarAlt = sonarRead();
+        sonarAlt = sonarCalculateAltitude(sonarAlt, getCosTiltAngle());
```

Comment:
Why split the calculation into two lines but not two variables? It's the stack after all, and will likely get optimized.